hibernate-rx-core/src/main/java/org/hibernate/rx/loader/RxOuterJoinLoader.java

```diff
@@ -0,0 +1,383 @@
+				if ( collectionPersister.isArray() ) {
+					//for arrays, we should end the collection load before resolving
+					//the entities, since the actual array instances are not instantiated
+					//during loading
+					//TODO: or we could do this polymorphically, and have two
+					//      different operations implemented differently for arrays
+					endCollectionLoad( resultSetId, session, collectionPersister );
+				}
+			}
+		}
+		return collectionPersisters;
+	}
+
+	// Added this just to catch the exception and make the code more readable
+	private CompletionStage<Object> rxRows(ResultSet rs, QueryParameters queryParameters, SharedSessionContractImplementor session, boolean returnProxies, ResultTransformer forcedResultTransformer, EntityKey optionalObjectKey, LockMode[] lockModesArray, boolean createSubselects, List subselectResultKeys, List hydratedObjects, EntityKey[] keys) {
+		try {
+			return getRxRowFromResultSet( rs, session, queryParameters, lockModesArray, optionalObjectKey, hydratedObjects,
+					keys, returnProxies, forcedResultTransformer
+			);
+		} catch (SQLException sqle) {
+			throw getFactory().getJdbcServices().getSqlExceptionHelper().convert(
+					sqle,
+					"could not load an entity batch: " + MessageHelper.infoString(
+							getEntityPersisters()[0],
+							queryParameters.getOptionalId(),
+							session.getFactory()
+					),
+					sql
+			);
+		}
+	}
+
+	private CompletionStage<Object> getRxRowFromResultSet(
+			final ResultSet resultSet,
+			final SharedSessionContractImplementor session,
+			final QueryParameters queryParameters,
+			final LockMode[] lockModesArray,
+			final EntityKey optionalObjectKey,
+			final List hydratedObjects,
+			final EntityKey[] keys,
+			boolean returnProxies,
+			ResultTransformer forcedResultTransformer) throws SQLException, HibernateException {
+		final Loadable[] persisters = getEntityPersisters();
+		final int entitySpan = persisters.length;
+		extractKeysFromResultSet( persisters, queryParameters, resultSet, session, keys, lockModesArray, hydratedObjects );
+
+//		FIXME: for later
+		registerNonExists( keys, persisters, session );
+
+		// this call is side-effecty
+		return getRxRow( resultSet, persisters, keys, queryParameters, optionalObjectKey, lockModesArray, hydratedObjects, session )
+				.thenApply( row -> {
+
+			//		TODO: We don't do collection of elements for now
+			//		readCollectionElements( row, resultSet, session );
+
+					if ( returnProxies ) {
+						// now get an existing proxy for each row element (if there is one)
+						final PersistenceContext persistenceContext = session.getPersistenceContextInternal();
+						for ( int i = 0; i < entitySpan; i++ ) {
+							Object entity = row[i];
+							Object proxy = persistenceContext.proxyFor( persisters[i], keys[i], entity );
+							if ( entity != proxy ) {
+								// force the proxy to resolve itself
+								( (HibernateProxy) proxy ).getHibernateLazyInitializer().setImplementation( entity );
```

Comment:
@DavideD how come this isn't just:

    ( (HibernateProxy) proxy ).getHibernateLazyInitializer().initialize();

Reply:
I don't know, that's how it's done in Loader#getRowFromResultSet in ORM